demo_modules/slideshow/interfaces.js

```diff
@@ -50,16 +80,15 @@ export class ClientLoadStrategy {
 }
 
 export class ServerDisplayStrategy {
-  init() {
-    // Return a promise when initialization is complete.
-    return Promise.resolve();
-  }
   tick(time, delta) {
     // Coordinate with the clients about what should be shown.
   }
-  newContent(content) {
-    // A notification from the load strategy that new content has been
-    // discovered. The parameter is an array of content identifiers.
+  clipRectForMetadata(metadata, client) {
+    // Returns a clipping rect for the content specified by the metadata for
+    // the specified client. The returned Rectangle should be in the unit-space
+    // of the content (0,0) contentWidth x contentHeight. The strategy can
+    // instead return null to denote that no clipping should occur.
+    return null;
```

Comment:
Why not just return the metadata extents in this case? Boo to null, I say.

Reply:
Each of these APIs gives a different strategy the option of giving up on any splitting. Here, we allow the loading strategy to say, you know what, you don't really want this metadata.